api/permissioned.go

```diff
@@ -41,6 +41,12 @@ func PermissionedWorkerAPI(a Worker) Worker {
 	return &out
 }
 
+func PermissionedAPI[T any](a T) T {
```

Comment:
yay generics exist now; We could drop the other ones now, but that's a separate PR, not now